### PR TITLE
API guide 复杂网络

### PR DESCRIPTION
**Pull request by tensor-tang**
solve #303 

预览效果：

image

### CODE REVIEW

File: doc/fluid/api/api_guides/low_level/nets.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+..  _api_guide_nets:
+
+###########
+复杂网络
+###########
+
+在处理复杂功能时，我们通常需要写大量的代码来构建比较复杂的 `神经网络 <https://zh.wikipedia.org/wiki/人工神经网络>`_ 。
+因此，为了方便用户更加容易的搭建复杂网络模型，我们提供了一些比较常用的基本函数模块，以此来简化用户的代码量，从而降低开发成本。
+这些模块通常只是由细粒度的函数由一定的逻辑拼接组合而成，实现代码请参考 `nets.py <https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/nets.py>`_ 。
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/nets.rst]: - 第7行，去掉“比较”
- 第8行，容易的-》容易地
- 第9行，这些模块通常是由细粒度的函数根据一定的逻辑拼接组合而成

File: doc/fluid/api/api_guides/low_level/nets.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+1.simple_img_conv_pool
+----------------------
+
+:code:`simple_img_conv_pool` 是由 :code:`Convolution2d` 与 :code:`Pool2d` 串联而成。
+该模块在图像分类模型中广泛使用，比如应用在处理 `MNIST <https://en.wikipedia.org/wiki/MNIST_database>`_ 数字分类的问题。
+关于 :code:`Convolution2d` 可以参考 :ref:`api_fluid_layers_conv2d`, :code:`Pool2d` 可以参考 :ref:`api_fluid_layers_pool2d`。
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/nets.rst]: - 第15行去掉“处理”
- 第16行去掉，两个跳转链接放到14行: 是由 :ref:`api_fluid_layers_conv2d` 与 :ref:`api_fluid_layers_pool2d ` 串联而成。全文同

File: doc/fluid/api/api_guides/low_level/nets.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+
+2.img_conv_group
+----------------
+
+:code:`img_conv_group` 是由 :code:`Convolution2d`, :code:`BatchNorm`, :code:`DropOut` 和 :code:`Pool2d` 组成。
+可以实现多个 :code:`Convolution2d`, :code:`BatchNorm` 和 :code:`DropOut` 的串联单元与一个 :code:`Pool2d` 组合的模块。
+:code:`Convolution2d`, :code:`BatchNorm` 和 :code:`DropOut` 的数量都可以分别控制，从而得到多样的组合。
+该模块通常广泛使用在比较复杂的图像分类任务中，比如 `VGG16 <https://github.com/PaddlePaddle/Paddle/blob/develop/benchmark/fluid/models/vgg.py#L30>`_ 。
+关于 :code:`Convolution2d` 可以参考 :ref:`api_fluid_layers_conv2d`, :code:`BatchNorm` 可以参考 :ref:`api_fluid_layers_batch_norm`, :code:`DropOut` 可以参考 :ref:`api_fluid_layers_dropout`, :code:`Pool2d` 可以参考 :ref:`api_fluid_layers_pool2d`。
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/nets.rst]: 28行去掉，跳转链接放到24行。全文同

File: doc/fluid/api/api_guides/low_level/nets.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+API Reference 请参考 :ref:`api_fluid_nets_simple_img_conv_pool`
+
+
+2.img_conv_group
+----------------
+
+:code:`img_conv_group` 是由 :code:`Convolution2d`, :code:`BatchNorm`, :code:`DropOut` 和 :code:`Pool2d` 组成。
+可以实现多个 :code:`Convolution2d`, :code:`BatchNorm` 和 :code:`DropOut` 的串联单元与一个 :code:`Pool2d` 组合的模块。
+:code:`Convolution2d`, :code:`BatchNorm` 和 :code:`DropOut` 的数量都可以分别控制，从而得到多样的组合。
+该模块通常广泛使用在比较复杂的图像分类任务中，比如 `VGG16 <https://github.com/PaddlePaddle/Paddle/blob/develop/benchmark/fluid/models/vgg.py#L30>`_ 。
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/nets.rst]: - 通常/广泛，两个词重复，去掉一个
- VGG16不要加源码路径，源码常常会修改，导致难以维护。可以加论文或者其他链接。

File: doc/fluid/api/api_guides/low_level/nets.rst [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+API Reference 请参考 :ref:`api_fluid_nets_simple_img_conv_pool`
+
+
+2.img_conv_group
+----------------
+
+:code:`img_conv_group` 是由 :code:`Convolution2d`, :code:`BatchNorm`, :code:`DropOut` 和 :code:`Pool2d` 组成。
+可以实现多个 :code:`Convolution2d`, :code:`BatchNorm` 和 :code:`DropOut` 的串联单元与一个 :code:`Pool2d` 组合的模块。
+:code:`Convolution2d`, :code:`BatchNorm` 和 :code:`DropOut` 的数量都可以分别控制，从而得到多样的组合。
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/nets.rst]: - 25行，缺少主语
- 26行，“其中” :code:`Convolution2d`, :code:`BatchNorm` 和 :code:`DropOut` 的数量都可以分别控制，从而得到多样的组合。
- Convolution2d-》Conv2d，下同

File: doc/fluid/api/api_guides/low_level/nets.rst [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+API Reference 请参考 :ref:`api_fluid_nets_img_conv_group`
+
+
+3.sequence_conv_pool
+--------------------
+
+:code:`sequence_conv_pool` 是由 :code:`sequence_conv` 与 :code:`sequence_pool` 串联而成。
+该模块在 `自然语言处理 <https://zh.wikipedia.org/wiki/自然语言处理>`_ 以及 `语音识别 <https://zh.wikipedia.org/wiki/语音识别>`_ 等领域均有广泛应用，
+比如 `文本分类模型 <https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleNLP/text_classification/nets.py>`_ , 
+`TagSpace <https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleRec/tagspace/train.py>`_  以及 `Multi-view Simnet <https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleRec/multiview_simnet/nets.py>`_ 等模型。
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/nets.rst:37]: TagSpace，Multi-view Simnet有中文么？

File: doc/fluid/api/api_guides/low_level/nets.rst [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+API Reference 请参考 :ref:`api_fluid_nets_glu`
+
+
+5.scaled_dot_product_attention
+------------------------------
+:code:`scaled_dot_product_attention` 来源于论文 `Attention Is All You Need <https://arxiv.org/pdf/1706.03762.pdf>`_ ，主要是由 :code:`fc` 和 :code:`softmax` 组成。
+对于输入数据 :code:`Queries` , :code:`Key` 和 :code:`Values` 按照如下公式求出 :code:`Attention` 。
+
+.. math::
+ Attention(Q, K, V)= softmax(QK^\mathrm{T})V
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/nets.rst:56]: 预览图里没有公式及后面的截图，请补充

File: doc/fluid/api/api_guides/low_level/nets.rst (hunk 1 above)
Comment (tensor-tang) [reply doc/fluid/api/api_guides/low_level/nets.rst]: OK，thx.

File: doc/fluid/api/api_guides/low_level/nets.rst (hunk 2 above)
Comment (tensor-tang) [reply doc/fluid/api/api_guides/low_level/nets.rst]: OK, thx

File: doc/fluid/api/api_guides/low_level/nets.rst (hunk 5 above)
Comment (tensor-tang) [reply doc/fluid/api/api_guides/low_level/nets.rst]: OK, thx

File: doc/fluid/api/api_guides/low_level/nets.rst (hunk 4 above)
Comment (tensor-tang) [reply doc/fluid/api/api_guides/low_level/nets.rst]: > VGG16不要加源码路径，源码常常会修改，导致难以维护。可以加论文或者其他链接。

OK, 本来的考虑是模型已经很稳定了，应该不会经常改动。不过也可以换成论文。

File: doc/fluid/api/api_guides/low_level/nets.rst (hunk 3 above)
Comment (tensor-tang) [reply doc/fluid/api/api_guides/low_level/nets.rst]: OK ， thx

File: doc/fluid/api/api_guides/low_level/nets.rst (hunk 6 above)
Comment (tensor-tang) [reply doc/fluid/api/api_guides/low_level/nets.rst:37]: 貌似没有，参考链接都是models里面的同名，方便用户参考学习。

File: doc/fluid/api/api_guides/low_level/nets.rst (hunk 7 above)
Comment (tensor-tang) [reply doc/fluid/api/api_guides/low_level/nets.rst:56]: 好的。

**Review by tensor-tang — COMMENTED** [review]
Done, Thanks

image

File: doc/fluid/api/api_guides/low_level/nets.rst [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+该模块在图像分类模型中广泛使用，比如应用在 `MNIST <https://en.wikipedia.org/wiki/MNIST_database>`_ 数字分类的问题。
+
+API Reference 请参考 :ref:`api_fluid_nets_simple_img_conv_pool`
+
+
+2.img_conv_group
+----------------
+
+:code:`img_conv_group` 是由 :ref:`api_fluid_layers_conv2d` , :ref:`api_fluid_layers_batch_norm`, :ref:`api_fluid_layers_dropout` 和 :ref:`api_fluid_layers_pool2d` 组成。
+该模块可以实现多个 :ref:`api_fluid_layers_conv2d` , :ref:`api_fluid_layers_batch_norm` 和 :ref:`api_fluid_layers_dropout` 的串联单元与一个 :ref:`api_fluid_layers_pool2d` 组合的模块。
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/nets.rst]: 24行，该模块可以实现……模块，（模块可以实现模块）不是很通顺。

Comment (tensor-tang) [reply doc/fluid/api/api_guides/low_level/nets.rst]: Thanks， 已改正

**Comment by luotao1** [comment]
LGTM

**Comment by luotao1** [comment]
@tink2123 @shanyi15 请检查格式

**Review by shanyi15 — APPROVED** [review]
格式LGTM， thanks!

**Comment by tink2123** [comment]
LGTM
